gui/js/measurement-tools-tab.js

```diff
@@ -1,29 +1,52 @@
-init_slider("set_widget_scale", 0.2, 2.0, 0.01, [0.6]);
-
-function add_measurement_tool(name, icon) {
-    const area = $('#measurement-tools');
-    area.append(`<div class='col-4 center' style='padding: 0 5px'>
-                    <label style="width: 100%; height: 100%">
-                        <input id='set_tool_${icon}' type='checkbox' name='measurement-tool' class='radio-button' />
-                        <div class='block btn glass'>
-                            <i style="font-size: 60px" class='material-icons'>${icon}</i> 
-                            <br> 
-                            <span>${name}</span>
-                        </div>
-                    </label>
-                </div>`);
-
-    $('#set_tool_' + icon).change(function () {
-        $('#measurement-tools .radio-button').not('#set_tool_' + icon).prop('checked', false);
-
-        if (this.checked) {
-            window.call_native('set_measurement_tool', name);
-        } else {
-            window.call_native('set_measurement_tool', 'none');
-        }
-    });
+class MeasurementToolApi extends IApi {
+    name = 'measurement_tool';
+
+    /**
+     * TODO
+     * @param name {string}
+     * @param icon {string}
+     */
+    addMeasurementTool(name, icon) {
+        const area = document.getElementById('measurement-tools');
+
+        const tool = CosmoScout.loadTemplateContent('measurement-tools');
+
+        tool.innerHTML = tool.innerHTML
+            .replace(this.regex('CONTENT'), name)
+            .replace(this.regex('ICON'), icon)
+            .trim();
+
+        tool.addEventListener('click', () => {
+            CosmoScout.callNative('set_celestial_body', name);
+        });
+
+        tool.addEventListener('change', (event) => {
+            document.querySelectorAll('#measurement-tools .radio-button').forEach((node) => {
+                if (node.id !== `set_tool_${icon}`) {
+                    node.checked = false;
+                }
+            });
+
+            if (event.target.checked) {
+                CosmoScout.callNative('set_measurement_tool', name);
+            } else {
+                CosmoScout.callNative('set_measurement_tool', 'none');
+            }
+        });
+
+        area.appendChild(tool);
+    }
+
+    /**
+     * Deselect all measurement tools
+     */
+    deselectMeasurementTool() {
```

Comment:
This could be simply `deselect`.

Reply:
Done in https://github.com/cosmoscout/csp-measurement-tools/pull/3/commits/d3300a41f7b9e142a4c56b97d459abe450f4fcb7